projects/kit/components/pdf-viewer/pdf-viewer.service.ts

```diff
@@ -16,4 +20,19 @@ export class TuiPdfViewerService extends AbstractTuiDialogService<
 > {
     protected readonly component = DIALOG;
     protected readonly defaultOptions: TuiPdfViewerOptions<any> = DEFAULT_OPTIONS as any;
+
+    open<G>(
+        content: () => SafeResourceUrl,
+        options: Partial<TuiPdfViewerOptions<any>>,
+    ): Observable<G>;
+    open<G>(
+        content: Content<G>,
+        options: Partial<TuiPdfViewerOptions<any>>,
+    ): Observable<G>;
+    open<G>(
+        content: (() => SafeResourceUrl) | Content<G>,
+        options: Partial<TuiPdfViewerOptions<any>> = {},
```

Comment:
can we simplify overrides?

```ts
type PolymorpheusContent<G, T> = PolymorpheusContent<TuiBaseDialogContext<G> & TuiPdfViewerOptions<T>>
type Content<G> = (() => SafeResourceUrl) | PolymorpheusContent<G>;


/// .... and then use only


open<G, T>(content: Content<G, T>, options: Partial<TuiPdfViewerOptions<T>> = {}} {
  return super.open(content, options);
}
```

Reply:
Right. We don't even need the overrides )